spec/services/orders/handle_fees_service_spec.rb

```diff
@@ -4,24 +4,263 @@
 
 RSpec.describe Orders::HandleFeesService do
   let(:order_cycle) { create(:order_cycle) }
-  let(:order) { create(:order_with_line_items, line_items_count: 1, order_cycle:) }
+  let(:order) {
+    create(:order_with_line_items, line_items_count: 1, order_cycle:,
+                                   distributor: order_cycle.distributors.first)
+  }
   let(:line_item) { order.line_items.first }
 
   let(:service) { Orders::HandleFeesService.new(order) }
   let(:calculator) {
-    double(OpenFoodNetwork::EnterpriseFeeCalculator, create_order_adjustments_for: true)
+    instance_double(OpenFoodNetwork::EnterpriseFeeCalculator, create_order_adjustments_for: true)
```

Comment:
Given that the specs already have a distributor and order cycle record, I wonder if it wouldn't hurt to just use a real EnterpriseFeeCalculator.

```suggestion
    OpenFoodNetwork::EnterpriseFeeCalculator.new(order_cycle.distributors.first, order_cycle)
```

It would then be easier to remove some of the mocks below and use data instead.

Reply:
This was the original testing approach, so I kept it. I think it's fine to mock here given the `EnterpriseFeeCalculator` is something we wrote, and by using an instance double we have additional checks to make sure mocked method do exist on the real implementation.
Technically if you use a real `EnterpriseFeeCalculator` it becomes integration testing.

Reply:
Good points.

> Technically if you use a real `EnterpriseFeeCalculator` it becomes integration testing.

Would that be a bad thing? I'm not sure we have a rule that all `services` specs must be pure unit tests. I thought that integration tests would be more valuable.